src/utils/twoFactor.js

```diff
@@ -0,0 +1,181 @@
+    } else {
+        return await serverResponse
+    }
+}
+
+export const twoFactorRequest = async (wallet, account, request) => {
+    if (wallet.tempTwoFactorAccount) account = wallet.tempTwoFactorAccount
+    const { accountId } = account
+    console.log('makeTwoFactorRequest', accountId, request, account)
+    const contract = getContract(account)
+    await deleteUnconfirmedRequests(contract)
+    const request_id = await getNextRequestId(contract)
+    await addRequestAndConfirm(contract, request)
+    const request_id_after = await getNextRequestId(contract)
+    if (request_id_after > request_id) {
+        const data = { request_id, request }
+        const method = await wallet.get2faMethod()
+        return await sendTwoFactorRequest(wallet, accountId, method, request_id, data)
+    }
+}
+
+export const twoFactorAddKey = (wallet, account, publicKey, contractId, fullAccess = false) => {
+    const request = {
+        receiver_id: account.accountId,
+        actions: [addKeyAction(account, publicKey, contractId, fullAccess)]
+    }
+    return twoFactorRequest(wallet, account, request)
+}
+
+export const twoFactorRemoveKey = (wallet, account, publicKey) => {
+    const request = {
+        receiver_id: account.accountId,
+        actions: [deleteKeyAction(publicKey)]
+    }
+    return twoFactorRequest(wallet, account, request)
+}
+/********************************
+Deploys 2/3 multisig contract with keys from contract-helper and localStorage
+@todo check account has enough near to actually do this
+********************************/
+export const twoFactorDeploy = async (wallet) => {
+    const accountData = await wallet.loadAccount()
+    // get multisig contract to deploy
+    const contractBytes = new Uint8Array(await fetch('/multisig.wasm').then((res) => res.arrayBuffer()))
+    // get account
+    const { accountId } = accountData
+    const account = wallet.getAccount(accountId)
+    // 1. get localstorage keys
+    const accountKeys = await account.getAccessKeys();
+    // 2. get recovery method keys
+    const recoveryMethods = await wallet.getRecoveryMethods()
+    const recoveryKeysED = recoveryMethods.data.map((rm) => rm.publicKey)
+    // 3. get recovery keys that are NOT seed phrases && NOT null publicKey
+    const fak2lak = recoveryMethods.data.filter(({ kind, publicKey }) => kind !== 'phrase' && publicKey !== null).map((rm) => toPK(rm.publicKey))
+    // 4. push localStorage keys that are NOT recovery keys
+    fak2lak.push(...accountKeys.filter((ak) => !recoveryKeysED.includes(ak.public_key)).map((ak) => toPK(ak.public_key)))
+    // these are the keys we need to convert to limited access keys
+    // 5. get the server public key for wallet accountId (confirmOnlyKey)
+    const getAccessKey = await wallet.postSignedJson('/2fa/getAccessKey', {
+        accountId
+    })
+    if (!getAccessKey || !getAccessKey.success || !getAccessKey.publicKey) {
+        console.log('error getting publicKey from contract-helper')
```

Comment:
Should this return some error to caller?

Reply:
makes sense, throwing now.